### ISSUE TITLE
Warning: Received 'true' for a non-boolean attribute 'centered' on Modal when using defaultProps

### ISSUE THREAD

**Issue by Vahor** [mantinedev/mantine#3702]
### What package has an issue

@mantine/core

### Describe the bug

Since v6, Spotlight use a modal under the hood, this bug also occurs when using this package.

When I use `<Modal centered>` there are no issue. 
However, when I want to have these props shared for all modals using the MantineProvider:
```tsx
<MantineProvider
      theme={{
        components: {
          Modal: {
            defaultProps: {
              centered: true,
              withCloseButton: false
            }
          }
        }
      }}
    >
```
I get a warning saying `Warning: Received 'true' for a non-boolean attribute 'centered'.` (same issue with other boolean props)

This bug is not urgent as this is only a warning, in my case the modal is indeed centered.

### What version of @mantine/hooks page do you have in package.json?

6.0

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/snowy-fire-flgj0b?file=/src/App.tsx

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by cyantree** [comment]
The issue seems to be the following:

`<ModalBase __staticSelector="Modal"` tells `ModalBase` to use the theme defaults for `Modal`.
https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/Modal/ModalRoot/ModalRoot.tsx#L55

As in many other components some props will be extracted and the rest merged in `others`.
https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/ModalBase/ModalBase.tsx#L146

Because `others` gets spread on the DOM node these unsupported props will also be passed to the DOM resulting in the described error.
https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/ModalBase/ModalBase.tsx#L206

Adding `centered` in the prop extraction won't work nicely because this component doesn't know it. So it needs to be defined in the types to get known.

This issue might apply to other props and also to other components.
I think these spreads are quite tricky but I don't have any idea for a better approach yet. Adding a filter method to only pass supported DOM props might be a way but this would also only mask the underlying issue that is that unsupported props might leak into places they shouldn't be.

But for the time being stripping the extra property in `ModalBase` might be the easiest thing to do.

**Comment by strblr** [comment]
Same problem here.

**Comment by rtivital** [comment]
Fixed in 6.0.3, thanks @cyantree!
